src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/ListBuilder.java

```diff
@@ -36,14 +36,15 @@
 
 /**
  * A utility class for building nested HTML lists. This class is implemented as a
- * stack of nested list elements where list items are added to the inner-most
- * list. Lists can be added to and removed from the stack using the {@code pushNested}
- * and {@code popNested} methods.
+ * stack of nested list/item pairs where list items are added to the inner-most
+ * list and nested lists are added to the current list item of the inner-most list.
+ * Lists can be added to and removed from the stack using the {@link #pushNestedList}
+ * and {@link #popNestedList} methods.
  */
 public class ListBuilder extends Content {
 
     private final HtmlTree root;
-    private final Deque<HtmlTree> stack = new ArrayDeque<>();
+    private final Deque<HtmlTree[]> stack = new ArrayDeque<>();
```

Comment:
Instead of an array of fixed-size 2, consider using a locally defined record, which means you get nice helpful names for the two components, instead of just `[0]` and `[1]`